Source/WebCore/dom/SlotAssignment.cpp

```diff
@@ -543,10 +546,11 @@ void ManualSlotAssignment::slotManualAssignmentDidChange(HTMLSlotElement& slot,
 void ManualSlotAssignment::didRemoveManuallyAssignedNode(HTMLSlotElement& slot, const Node& node, ShadowRoot& shadowRoot)
 {
     ASSERT(slot.containingShadowRoot() == &shadowRoot);
-    ASSERT_UNUSED(node, node.parentNode() == shadowRoot.host());
+    RefPtr shadowRootHost = shadowRoot.host();
```

Comment:
Since this is immediately unconditionally dereferenced I think we should use `Ref` instead of `RefPtr`. What do you think?